test/arcade/patch/agent/cell/PatchCellContainerTest.java

```diff
@@ -0,0 +1,144 @@
+                    cellID,
+                    cellParent,
+                    cellPop,
+                    cellAge,
+                    cellDivisions,
+                    cellState,
+                    cellVolume,
+                    cellHeight,
+                    cellCriticalVolume,
+                    cellCriticalHeight);
+
+    static class PatchCellMock extends PatchCellTissue {
+        PatchCellMock(PatchCellContainer container, Location location, Parameters parameters) {
+            super(container, location, parameters, null);
+        }
+
+        @Override
+        public PatchCellContainer make(int newID, CellState newState, MersenneTwisterFast random) {
+            return new PatchCellContainer(
+                    newID,
+                    id,
+                    pop,
+                    age,
+                    divisions,
+                    newState,
+                    volume,
+                    height,
+                    criticalVolume,
+                    criticalHeight);
+        }
+    }
+
+    @BeforeAll
+    public static void setupMocks() {
+        simMock = mock(PatchSimulation.class);
+        locationMock = mock(PatchLocation.class);
+        parametersMock = spy(new Parameters(new MiniBox(), null, null));
+        randomMock = mock(MersenneTwisterFast.class);
+    }
+
+    @Test
+    public void setState_proliferateWithIC_drawsFromDistribution() {
+        Schedule scheduleMock = mock(Schedule.class);
+        doReturn(cellID + 1).when(simMock).getID();
+        doReturn(scheduleMock).when(simMock).getSchedule();
+        doReturn(null).when(scheduleMock).scheduleRepeating(anyInt(), anyInt(), any());
+        PatchCellFactory factoryMock = mock(PatchCellFactory.class);
+        doReturn(null).when(factoryMock).getLinks(anyInt());
+        doReturn(factoryMock).when(simMock).getCellFactory();
+        doReturn(0.5).when(randomMock).nextDouble();
+
+        MiniBox population = new MiniBox();
+        double volume = 500;
+        population.put("CLASS", "tissue");
+        population.put("COMPRESSION_TOLERANCE", 0);
+        population.put("(DISTRIBUTION)/NECROTIC_FRACTION", "NORMAL");
+        population.put("NECROTIC_FRACTION_MU", 0.5);
+        population.put("NECROTIC_FRACTION_SIGMA", 0.1);
+        population.put("NECROTIC_FRACTION_IC", "MU");
+        population.put("proliferation/SYNTHESIS_DURATION", 1);
+        population.put("SENESCENT_FRACTION", 0);
+        population.put("ENERGY_THRESHOLD", 0);
+        population.put("APOPTOSIS_AGE", 0);
+        population.put("ACCURACY", 0);
+        population.put("AFFINITY", 0);
+        population.put("DIVISION_POTENTIAL", 50);
+        population.put("MAX_DENSITY", 0);
+        population.put("CAR_ANTIGENS", 0);
+        population.put("SELF_TARGETS", 0);
+        doReturn(population).when(factoryMock).getParameters(anyInt());
+        Parameters parameters = new Parameters(population, null, randomMock);
```

Comment:
So this is the important test - you need to make sure the daughter cells inherit the original distribution and not the degenerate distribution. This is how this happens in `PatchModuleProliferation`:

```java

CellContainer newContainer = cell.make(newID, State.UNDEFINED, random);
Parameters newParameters = cell.getParameters();
PatchCell newCell =
        (PatchCell)
                newContainer.convert(
                        sim.getCellFactory(),
                        newLocation,
                        random,
                        newParameters);

```

This test isn't actually testing how new cells are created, which is related to @allison-li-1016's question.